File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        //// Linear Search O(n)
    //    int n = nums.size();
    //    for(int i = 0; i<n ; i++){
    //         if(nums[i] == target){
    //             return i;
    //         }
    //    } 
    //    return -1;



    //// this is Binary Search O(log n)
    int st =0 , end = nums.size();
    while(st<=end){
        int mid = st + (end-st)/2;
        if(nums[mid] == target){
            return mid;
        }else if(nums[mid] > target){
             end = mid-1;
        }else{
          st = mid +1; 
        }

    }
    return -1;
    }
};

File: Data Structures & Algorithms/find-peak-element/submission-3.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        n = len(nums)
        st = 0 
        end = n-1
        while st < end :
            mid = st + (end-st)//2
            if nums[mid] < nums[mid+1]:
                st = mid+1
            else:
                end = mid
        return st
        